app/src/main/java/com/jakdor/apapp/ui/apartment/ApartmentFragment.kt

```diff
@@ -0,0 +1,258 @@
+            override fun afterTextChanged(s: Editable?) {
+
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val apartmentName = apartment_name_editText.text.toString()
+
+                viewModel?.apartmentNameValidation(apartmentName)
+            }
+
+        })
+
+        apartment_city_editText.addTextChangedListener(object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val apartmentCity = apartment_city_editText.text.toString()
+
+                viewModel?.apartmentCityValidation(apartmentCity)
+            }
+
+        })
+
+        apartment_street_editText.addTextChangedListener(object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val apartmentStreet = apartment_street_editText.text.toString()
+
+                viewModel?.apartmentStreetValidation(apartmentStreet)
+            }
+
+        })
+
+        apartment_number_editText.addTextChangedListener(object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val apartmentNumber = apartment_number_editText.text.toString()
+
+                viewModel?.apartmentNumberValidation(apartmentNumber)
+            }
+
+        })
+    }
+
+    fun onPhotosReturned(returnedPhotos: ArrayList<String>) {
+        photos.clear()
+            if(returnedPhotos.size > 0) {
+                for (image in returnedPhotos) {
+                    photos.add(Picture(image, BitmapFactory.decodeFile(image)))
+                }
+            }
+            recyclerViewAdapter.notifyDataSetChanged()
+            item_recycler.scrollToPosition(photos.size - 1)
+    }
+
+    fun observeApartmentIdLiveData(){
+        viewModel?.apartmentIdLiveData?.observe(this, Observer {
+            handleNewApartmentId(it)
+        })
+    }
+
+    fun handleNewApartmentId(apartmentId: Int){
+        Toast.makeText(activity,apartmentId.toString(),Toast.LENGTH_SHORT).show()
+    }
+
+    private fun observeApartmentNameStatus() {
+        viewModel?.apartmentNameStatus?.observe(this, Observer {
```

Comment:
Nie nadużywaj LiveData jak nie jak masz pewność, że rezultat z viewModelu przyjdzie synchronicznie. Taka ogólna uwaga do całego fragmentu, w rejestracji wszędzie było LiveData bo mogły przyjść jakieś uwaga asynchronicznie do konkretnego pola z backendu i wtedy to było wygodne. Warto się zastanowić czy tutaj też jest taka sytuacja, czy może jednak LiveData nie jest tutaj potrzebne.